scripts/inspect-components

```diff
@@ -0,0 +1,488 @@
+#!/usr/bin/env python3
+
+"""Inspect and analyze React components."""
+
+import argparse
+from collections import namedtuple
+from concurrent import futures
+import logging
+import operator
+import os
+import re
+import sys
+
+
+# Define global regular expressions.
+_definition_exp = re.compile(r"""
+    ([\w\._]+)\s*  # Component name.
+    =\s*  # Assignation.
+    React\.createClass  # The create class statement.
+    [\s\S]*?  # Any character.
+    (?=  # Until the following happens.
+        propTypes\s*  # Properties are found.
+        :\s*  # Property definitions.
+        \{  # Brace opening.
+        ([\s\S]*?)  # Any character.
+        (?=  # Until the following happens.
+            \}  # A close brace is encountered.
+        )  # End of condition.
+    )  # End of condition.
+""", re.VERBOSE | re.MULTILINE)
```

Comment:
So this is all just witchcraft, right?

Less glibly, thanks for breaking this out with all the comments.